library/src/scala/Tuple.scala

```diff
@@ -13,9 +13,9 @@ sealed trait Tuple extends Product {
     runtime.Tuples.toArray(this)
 
   /** Create a copy of this tuple as a List */
-  inline def toList: List[Union[this.type]] =
+  inline def toList[This >: this.type <: Tuple]: List[Union[This]] =
```

Comment:
This is not backward tasty compatible, unfortunately.

Reply:
Does it mean it should be merged after the next minor release only, or that we can't do that at all?

Reply:
We cannot do it at all.

Reply:
Too bad 🥲
We have the same problem for `Tuple.map` by the way: https://github.com/lampepfl/dotty/issues/15992#issuecomment-1243700382.

Reply:
how can we sensibly improve API's known to be broken? to we need to add an `.api2` method to Tuple which returns an object with all the fixed APIs?

Reply:
The only true way is not to introduce broken APIs in the first place ... which means more testing when they are introduced. Of course that's too late for what's already published.

It might be possible to pull off something once `@binaryAPI` is introduced. As we could keep the broken version as `@binaryAPI private[Tuple]` and introduce the fixed version as public.

Reply:
> As we could keep the broken version as @binaryAPI private[Tuple] and introduce the fixed version as public.

I thought when inlining that would fail the accessibility check, as it would already be resolved to the specific overload?

Reply:
Accessibility from tasty is already relaxed to a degree.

Reply:
Suggestion from Martin: deprecate `Tuple.toList` and replace it with a new `Tuple.toIterator` that has a type parameter.

Not sure what to do for `Tuple.map`.